scripts/generateTypeSummary.ps1

```diff
@@ -0,0 +1,106 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+<#
+.SYNOPSIS
+    Creates a type summary from DLL
+
+.DESCRIPTION
+    Takes a Microsoft.Graph DLL and prints types as a flat text file
+    - Selects types that fall under Microsoft.Graph namespace
+    - Drops version information from full generic type names
+    - Types are sorted as enums, interfaces and classes
+      - Each type of types are sorted among themselves by name
+    - Properties and methods are also sorted by their name
+      - For deterministic results, overloaded methods are also sorted by their argument list
+
+.PARAMETER dllPath
+    Full path to Microsoft.Graph.dll
+
+.PARAMETER outputPath
+    Full Path to type summary file
+
+.EXAMPLE
+    PS C:\> .\generateTypeSummary.ps1 -dllPath C:\test\Microsoft.Graph.dll -outputPath C:\test\typeSummary.txt
+#>
+
+Param(
+    [string]$dllPath,
+    [string]$outputPath
+)
+
+$twoSpaces = "  "
+$fourSpaces = "    "
+
+$assembly = [System.Reflection.Assembly]::LoadFrom($dllPath)
+$types = $assembly.GetTypes() |
+    Where-Object { $_.FullName.StartsWith("Microsoft.Graph") } |
```

Comment:
nit: StartsWith("Microsoft.Graph.Beta") 

Likely not necessary

Reply:
This is for namespaces of the types, so `Microsoft.Graph.Beta` would be wrong in this particular case.